modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/ImagePins.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {event, select, zoom, zoomIdentity, zoomTransform} from 'd3';
+import PropTypes from 'prop-types';
+import React, {useLayoutEffect, useRef} from 'react';
+
+import NavigationButtons from './NavigationButtons';
+import {
+	moveDown,
+	moveLeft,
+	moveRight,
+	moveUp,
+	zoomIn,
+	zoomOut,
+} from './NavigationsUtils';
+import ZoomController from './ZoomController';
+
+const ImagePins = ({
+	changedScale,
+	enablePanZoom,
+	execZoomIn,
+	image,
+	imageSettings,
+	navigationController,
+	resetZoom,
+	selectedOption,
+	setChangedScale,
+	setResetZoom,
+	setSelectedOption,
+	setZoomInHandler,
+	setZoomOutHandler,
+	spritemap,
+	zoomController,
+	zoomInHandler,
+	zoomOutHandler,
+}) => {
+	const handlers = useRef();
+	const container = useRef();
+	const panZoom = useRef();
+
+	const svg = useRef(null);
```

Comment:
Instead of naming a ref as `svg`, why not `svgRef` or `svgElementRef`?

Reply:
sure